### PR TITLE
[Doc] Add example to  keyboard housekeeping and some minor fixes

### PR DESCRIPTION
**Pull request by ForsakenRei**
## Description
Main change:
- Add an example of using `void housekeeping_task_user(void)` to turn off RGB light as there is not a builtin way like `RGB_MATRIX_TIMEOUT`.

Minor changes:
- Add a link to key override into advanced keycode doc where `shift + backspeace = delete` is achieved by a macro.
- Fix the name for Space65 R3 in its README.
- Other minor format fixes in key override doc.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by ForsakenRei** [comment]
The failed ones are 
- gray_studio/space65r3:mechmerlin ❌
- gray_studio/space65r3:spidey3 ❌
but I didn't see them in the keymap folder...?

File: docs/custom_quantum_functions.md [hunk 1]
```diff
@@ -202,14 +202,77 @@ This function gets called at the end of all QMK processing, before starting the
...
+And add those lines to your `keymap.c`.
+
+```c
+static uint32_t key_timer;           // timer for last keyboard activity, use 32bit value and function to make longer idle time possible
+static void refresh_rgb(void);       // refreshes the activity timer and RGB, invoke whenever any activity happens
+static void check_rgb_timeout(void); // checks if enough time has passed for RGB to timeout
+bool is_rgb_timeout = false;         // store if RGB has timed out or not in a boolean
+
+void refresh_rgb()
+{
```
Comment (drashna) [inline docs/custom_quantum_functions.md:225]: Examples should follow coding style guidelines. 

Eg: 
```suggestion
void refresh_rgb(void) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
> The failed ones are
> 
> * `gray_studio/space65r3:mechmerlin` ❌
> * `gray_studio/space65r3:spidey3` ❌
>   but I didn't see them in the keymap folder...?

It's because you edited the readme for the folder, it looks like.

And as long as the default keymap compiles fine, that is fine.

File: docs/custom_quantum_functions.md (hunk 1 above)
Comment (ForsakenRei) [reply docs/custom_quantum_functions.md]: Just made those changes, let me know if anything else I need change.

**Comment by ForsakenRei** [comment]
> > The failed ones are
> > 
> > * `gray_studio/space65r3:mechmerlin` ❌
> > * `gray_studio/space65r3:spidey3` ❌
> >   but I didn't see them in the keymap folder...?
> 
> It's because you edited the readme for the folder, it looks like.
> 
> And as long as the default keymap compiles fine, that is fine.

I see.

**Review by drashna — APPROVED** [review]

File: docs/custom_quantum_functions.md [hunk 2]
```diff
@@ -202,14 +202,70 @@ This function gets called at the end of all QMK processing, before starting the
 
 Similar to `matrix_scan_*`, these are called as often as the MCU can handle. To keep your board responsive, it's suggested to do as little as possible during these function calls, potentially throtting their behaviour if you do indeed require implementing something special.
 
+### Example `void housekeeping_task_user(void)` implementation
+
+This example will show you how to use `void housekeeping_task_user(void)` to turn off RGB light. For RGB Matrix just use `#define RGB_MATRIX_TIMEOUT` in your `config.h`.
+
+First add those lines to your user `config.h`.
```
Comment (waffle87) [inline docs/custom_quantum_functions.md:209]:
```suggestion
This example will show you how to use `void housekeeping_task_user(void)` to turn off [RGB Light](feature_rgblight.md). For RGB Matrix, the [builtin](https://docs.qmk.fm/#/feature_rgb_matrix?id=additional-configh-options) `RGB_MATRIX_TIMEOUT` should be used.

First, add the following lines to your keymap's `config.h`:
```

File: docs/custom_quantum_functions.md [hunk 3]
```diff
@@ -202,14 +202,70 @@ This function gets called at the end of all QMK processing, before starting the
...
+
+```c
+#define RGBLIGHT_SLEEP  // enable rgblight_suspend() and rgblight_wakeup() in keymap.c
+#define RGBLIGHT_TIMEOUT 900000  // ms to wait until rgblight time out, 900K ms is 15min.
+```
+
+And add those lines to your `keymap.c`.
```
Comment (waffle87) [inline docs/custom_quantum_functions.md:216]:
```suggestion
Next, add the following code to your `keymap.c`:
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**